turbo/engineapi/engine_block_downloader/block_downloader.go

```diff
@@ -126,11 +126,25 @@ func (e *EngineBlockDownloader) scheduleHeadersDownload(
 }
 
 // waitForEndOfHeadersDownload waits until the download of headers ends and returns the outcome.
-func (e *EngineBlockDownloader) waitForEndOfHeadersDownload() headerdownload.SyncStatus {
-	for e.hd.PosStatus() == headerdownload.Syncing {
-		time.Sleep(10 * time.Millisecond)
+func (e *EngineBlockDownloader) waitForEndOfHeadersDownload(ctx context.Context) (headerdownload.SyncStatus, error) {
+	ticker := time.NewTicker(10 * time.Millisecond)
+	defer ticker.Stop()
+
+	logEvery := time.NewTicker(30 * time.Second)
```

Comment:
FYI: also can use `libcommon.Sleep()` and `time.After()`

Reply:
also if change `10 * time.Millisecond` to `100 * time.Millisecond` probably nothing will change

Reply:
> also if change `10 * time.Millisecond` to `100 * time.Millisecond` probably nothing will change

yeah probably, but the original code used 10ms sleep, so I didn't want to change its semantics, just fix the cancellation + ticker log